### DISCUSSION TITLE
Auto Update Option?

### DISCUSSION THREAD

**Discussion by asip88** [seerge/g-helper#2919] [General]
Is there an ability to auto-update? The process of downloading, unzipping, replacing the .exe with these frequent updates is just very burdensome. I saw previously that there is some config to stop G-helper from checking for updates but I don't see that option anywhere.

I'm on 0.181 and every time it's said there was an update previously, it just launches Edge and downloads the file to my desktop. I'd love an auto-update option or at the very least the ability to not check for updates since they are too frequent.

**Comment by seerge** [comment]
@asip88 hello, there is an autoupdater. When new final release is available, it will prompt you to update with an alert window. If you click `No` it will remember that and won't bother you till next update. Otherwise - it will autodownload and replace old exe.

**Reply by asip88** [reply]
hmm....it doesn't operate that way for me. it just downloads the new .exe from my browser to my desktop/downloads folder. is there some config that needs to happen to get it to auto replace the old exe?

**Reply by seerge** [reply]
@asip88 do you keep exe in some strange location, or do you have some limited permissions in windows ? Cause what it does after is runs powershell command to unzip / replace old file and exists from the app.

Check for a log  at`%appdata%\GHelper`, it records it's actions during update there

**Reply by ke1ne** [reply]
> @asip88 hello, there is an autoupdater. When new final release is available, it will prompt you to update with an alert window. If you click `No` it will remember that and won't bother you till next update. Otherwise - it will autodownload and replace old exe.

Maybe, add some checkbox, like "update to minor versions"?

**Reply by seerge** [reply]
@ke1ne i absolutely intentionally don't want that :) as whole purpose of "non final" releases is to let enthusiasts use them and not to bother mainstream public

**Reply by ke1ne** [reply]
> @ke1ne i absolutely intentionally don't want that :) as whole purpose of "non final" releases is to let enthusiasts use them and not to bother mainstream public

Sure, not problem. Thanks for your your work!

**Comment by asip88** [comment]
It does not prompt me to update again if I say no but the auto updated does
not work as stated. When I click yes it either does nothing or opens my
browser and downloads the new exe to my downloads location (desktop for
me). I don’t have my g-helper in any weird spot — it’s located in program
files folder

**Reply by seerge** [reply]
@asip88 

> It does not prompt me to update again if I say no

It will prompt for the next release, but it obviously remembers your choice for that specific release. 

>I don’t have my g-helper in any weird spot — it’s located in program
files folder

Actually `Program Files` is a system folder, that requires admin rights to write into that folder (i.e. unzip / replace old exe). So if you don't run app as admin currently - this could be a reason it doesn't update.

You can either start app as admin (and uncheck / check run on startup checkbox to ensure it will run as admin all the time with system) OR keep GHelper.exe in some other folder :)

**Reply by Graybooked** [reply]
> @asip88
> 
> > It does not prompt me to update again if I say no
> 
> It will prompt for the next release, but it obviously remembers your choice for that specific release.
> 
> > I don’t have my g-helper in any weird spot — it’s located in program
> > files folder
> 
> Actually `Program Files` is a system folder, that requires admin rights to write into that folder (i.e. unzip / replace old exe). So if you don't run app as admin currently - this could be a reason it doesn't update.
> 
> You can either start app as admin (and uncheck / check run on startup checkbox to ensure it will run as admin all the time with system) OR keep GHelper.exe in some other folder :)

The end user should be told this explicitly in the program every update then.
